keypoints/include/pcl/keypoints/impl/harris_3d.hpp

```diff
@@ -508,7 +508,7 @@ pcl::HarrisKeypoint3D<PointInT, PointOutT, NormalT>::refineCorners (PointCloudOu
   Eigen::Matrix3f NNT;
   Eigen::Matrix3f NNTInv;
   Eigen::Vector3f NNTp;
-  float diff;
+  float diff = 0.;
```

Comment:
Is this really the best solution? `diff` is overwritten immediately, and there is anyway no reason to have this variable. How about removing it completely and using this at the end of the do-while loop:
```c++
      if((corners[cIdx].getVector3fMap () - corner.getVector3fMap()).squaredNorm () <= 1e-6) break;
    } while (++iterations < max_iterations);
```